### PR TITLE
Update connect function in api.py to accept proxies

### PR DESCRIPTION
**Pull request by manojbalaji1**
In certain scenarios, druid url's are accessible explicitly through proxies, so added proxies as input parameter for connection which is passed to requests.post in Cursor

@donbowman @mistercrunch can you please review this PR?

### CODE REVIEW

File: pydruid/db/api.py [hunk 1]
```diff
@@ -21,14 +21,15 @@ class Type(object):
...
-    header=False,
+        host="localhost",
+        port=8082,
+        path="/druid/v2/sql/",
+        scheme="http",
+        user=None,
+        password=None,
+        context=None,
+        header=False,
+        proxies=None
```
Comment (mistercrunch) [inline pydruid/db/api.py]: trailing comma pls

File: pydruid/db/api.py [hunk 2]
```diff
@@ -105,19 +106,19 @@ def get_type(value):
...
+            self,
+            host="localhost",
+            port=8082,
+            path="/druid/v2/sql/",
+            scheme="http",
+            user=None,
+            password=None,
+            context=None,
+            header=False,
+            proxies=None
```
Comment (mistercrunch) [inline pydruid/db/api.py]: trailing comma

**Comment by mistercrunch** [comment]
`pre-commit install` as noted in README should help

File: pydruid/db/api.py (hunk 1 above)
Comment (manojbalaji1) [reply pydruid/db/api.py]: @mistercrunch Done!

File: pydruid/db/api.py (hunk 2 above)
Comment (manojbalaji1) [reply pydruid/db/api.py]: @mistercrunch done

**Comment by manojbalaji1** [comment]
> Trailing coma + rebase, otherwise LGTM

Completed requested changes

**Pull request merged**